packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -54,33 +56,37 @@ export async function getFileLink(
                 result = await getFileLinkCore(getToken, odspUrlParts, identityType, logger);
                 success = true;
             } catch (err) {
-                // If it is not retriable, then just return undefined
+                // If it is not retriable, then just throw
                 if (!canRetryOnError(err)) {
-                    fileLinkCache.remove(cacheKey);
-                    return undefined;
+                    throw err;
                 }
                 // If the error is throttling error, then wait for the specified time before retrying.
                 // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
                 retryAfterMs = getRetryDelayFromError(err) ?? Math.min(retryAfterMs * 2, 8000);
                 await delay(retryAfterMs);
             }
         } while (!success);
+
+        // We are guaranteed to run the getFileLinkCore at least once with successful result (which must be a string)
+        assert(result !== undefined, "Unexpected undefined result from getFileLinkCore");
         return result;
     };
     fileLinkCache.add(cacheKey, valueGenerator);
-    return fileLinkCache.get(cacheKey);
+    // PromiseCache secretly invokes the promise generating function when adding if the cache was empty, and adds the
+    // result promise to the cache.  So, this get() is guaranteed to return a result since we know our
+    // promise-generating function will return a string or else there was already a string result available.
+    const fileLink = fileLinkCache.get(cacheKey);
+    assert(fileLink !== undefined, "Unexpected empty share file link cache");
```

Comment:
This assert seems to be not needed (it essentially tests behavior of PromiseCache, which feels like should be done in UTs)

Reply:
It feels like fileLinkCache should not be changed - there is no way (with your changes) to have undefined values in it

Reply:
I was actually thinking about just removing the usage of `PromiseCache`, since a simple `Map` works just as well here.  WDYT?

Reply:
Did this in latest.

Reply:
I think the intention of PromiseCache is to clear cache after some time. I'm not sure if this is applicable here or not

Reply:
Here it was not applicable, no args passed to constructor: 

```ts
// Store cached responses for the lifetime of web session as file link remains the same for given file item
const fileLinkCache = new PromiseCache<string, string | undefined>();
```